### ISSUE TITLE
Schema options appear not to be working on subdocuments

### ISSUE THREAD

**Issue by lukesammy** [nestjs/mongoose#662]
## I'm submitting a...

[ ] Regression
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request

## Current behavior

Schema options passed into the Schema decorator on subdocuments don't do anything, i.e. do populate timestamps, remove _id etc.

<pre><code>
@Schema({ timestamps: true, _id: false }) <- These schema options don't do anything
export class Comment extends Types.Subdocument {
  @Prop()
  body: string;
}
</code></pre>

<pre><code>
Schema({ timestamps: true }) <- These schema options work fine
export class Card {
  @Prop()
  title: string;
  @Prop()
  body: string;
  @Prop([Comment])
  comments: Types.DocumentArray<Comment>;
}
</code></pre>

## Expected behavior

Schema options passed into the Schema decorator on subdocuments work as they do on the parent document.

## Minimal reproduction of the problem with instructions

Using the examples above we get the following

<pre><code>
{
    _id: ObjectId('5fbbeb9ab8c43da668cbd381'),
    title: 'test card',
    body: 'test body',
    comments: [
        {
            _id: ObjectId('5fbc06d2d43fa62222f10a71'),
            body: 'test comment'
        }
    ],
    createdAt: ISODate('2020-11-23T17:04:26.905Z'),
    updatedAt: ISODate('2020-11-23T19:00:34.016Z'),
    __v: 0
}
</code></pre>

## Environment

<pre><code>
Nest version: 7.0.0
Nest Mongoose: 7.0.2
 
For Tooling issues:
- Node version: 15.2.1
- Platform:  Linux

Others:
Yarn
</code></pre>

**Comment by kamilmysliwiec** [comment]
Fixed in 7.1.2

**Comment by amirqasemi74** [comment]
In #656 i request a feature. according to this issue so it is true that I can write that code like this?

```ts
@Schema()
class Salary {
    @Prop()
     type: string
}

// const SalarySchema = SchemaFactory.createForClass(Salary)

class Candidate {
//   @Prop({type: SalarySchema})
     @Prop(Salay)
     proposedSalay: Salary;
}

// or
class Candidate {
    //@Prop({type: SalarySchema})
     @Prop()
     proposedSalay: Salary;
}
```

I think documentation is needed to add.
